src/main/java/io/stargate/sgv2/jsonapi/api/model/command/deserializers/FilterClauseDeserializer.java

```diff
@@ -79,6 +77,16 @@ private void validate(List<ComparisonExpression> expressionList) {
   }
 
   private void validate(String path, FilterOperation<?> filterOperation) {
+    // First: $vector can only be used with $exists operator
+    if (path.equals(DocumentConstants.Fields.VECTOR_EMBEDDING_FIELD)
+        && ElementComparisonOperator.EXISTS != filterOperation.operator()) {
+      throw new JsonApiException(
+          ErrorCode.INVALID_FILTER_EXPRESSION,
+          String.format(
+              "Cannot filter on '%s' field using operator '%s': only '$exists' is supported",
```

Comment:
Do we have a test case for `findOne` with `$exists` for `$vector` field? Can we add one if it is not?

Reply:
Excellent question! I believe we do since the unit tests were failing for my first version. That's how I found out that `$exists` works on Vectors -- first version simply failed on every type of expression and lead to test failure(s).

So yes, this case is fortunately covered; I think @maheshrajamani originally added that test.